### ISSUE TITLE
Setting the highlighted code to empty string will not reflect the value change

### ISSUE THREAD

**Issue by FrancescoBorzi** [MurhafSousli/ngx-highlightjs#119]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] question
```

### OS and Version?
Ubuntu 18.04

### Versions
```
Angular CLI: 9.0.7
Node: 12.14.0
OS: linux x64

Angular: 9.0.6
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.3
@angular-devkit/build-angular     0.900.7
@angular-devkit/build-optimizer   0.900.7
@angular-devkit/build-webpack     0.900.7
@angular-devkit/core              9.0.3
@angular-devkit/schematics        9.0.7
@angular/cli                      9.0.7
@ngtools/webpack                  9.0.7
@schematics/angular               9.0.7
@schematics/update                0.900.7
rxjs                              6.5.4
typescript                        3.7.5
webpack                           4.41.2
```

### Description
I'm having a hard time updating from version 3.x to version 4.x as whenever in my app I change the code from a non-empty string to an empty string (`''`), the change will not be reflected

### Repro steps

I've created a minimal application that reproduces the issue:

1) Clone this app https://github.com/FrancescoBorzi/ngx-highlightjs-bug-reproduction
2) `npm i && ng serve`
3) Open the app and delete the content of the input. You'll see when the input is empty, the highlight is not being updated

```
<input [(ngModel)]="code">

<pre>
  <code [highlight]="code"></code>
</pre>
```

![reproduction](https://user-images.githubusercontent.com/75517/77111291-5fb22280-6a27-11ea-9094-9745c9ff2454.gif)

I tested on both versions `v4.0.2` and `v4.1.0-beta`, the bug is the same.

It worked correctly on version `v3.0.3`

### Desired functionality
Should correctly update the view displaying an empty string

### Mention any other details that might be useful

This bug appeared in 4.x, it worked fine in version 3.x

**Comment by MurhafSousli** [comment]
Please use stackblitz for reproduction

**Comment by FrancescoBorzi** [comment]
@MurhafSousli here you go: https://stackblitz.com/edit/ngx-highlightjs-sftnnx

try to delete the text from the input and you'll see the bug

I think removing this line `changes.code.currentValue &&` (line 55) will fix the issue

**Comment by FrancescoBorzi** [comment]
@MurhafSousli any news on this? were you able to reproduce the issue?

**Comment by MurhafSousli** [comment]
Here https://stackblitz.com/edit/ngx-highlightjs-yp5rvq?file=src/app/home/home.component.ts

It turns out that if you want bind an input, you need to turn of the line numbers feature

**Comment by FrancescoBorzi** [comment]
@MurhafSousli why closing this issue? that link of yours still has the same bug:

![image](https://user-images.githubusercontent.com/75517/77825816-0d01f600-710c-11ea-9235-7355d7769dd0.png)

![image](https://user-images.githubusercontent.com/75517/77825825-17bc8b00-710c-11ea-8be4-4e96b3a1c245.png)

![image](https://user-images.githubusercontent.com/75517/77825830-21de8980-710c-11ea-9db7-7c6eb86e5cee.png)

![image](https://user-images.githubusercontent.com/75517/77825833-260aa700-710c-11ea-8f2c-60b80655a8de.png)

**Comment by FrancescoBorzi** [comment]
@MurhafSousli 

![bug](https://user-images.githubusercontent.com/75517/77825982-f4dea680-710c-11ea-83e6-65c63e6ad77b.gif)

**Comment by Helias** [comment]
Confirmed.

**Comment by MurhafSousli** [comment]
Sorry, I didn't understand the issue at the beginning.

**Comment by FrancescoBorzi** [comment]
@MurhafSousli that way you are hacking it to be a space `' '` that is different than empty string `''`.

The bug is that `''` is falsy in js, so the condition `if (value)` will be false when `value` is `''`

**Comment by MurhafSousli** [comment]
Thats right, maybe we can check if the `value !== undefined` instead

**Comment by FrancescoBorzi** [comment]
@MurhafSousli yes, or even better just set to `''` whenever the value is `''` or `null` or `undefined`

**Comment by MurhafSousli** [comment]
Fixed in v4.1.0, feel free to reopen if you still have the issue!

**Comment by estanglerbm** [comment]
I think this fix was undone by commit d60dc10b5c5ce9d269214b2a7f6f6e9f3d02eaed (for issue #142), such as in 4.1.3 (for Angular 11).

I realize that you may want to have `<code highlight="">` do nothing (for one use case).  But need some way to clear the pieces that ngx-highlightjs created before the value went blank (in the other use case).

**Comment by MurhafSousli** [comment]
@estanglerbm can you provide a stackblitz to demonstrate the use case?

**Comment by estanglerbm** [comment]
https://stackblitz.com/edit/angular-ivy-ypchsm?file=src/app/app.component.html

If you click the yellow box for JSON or SQL, and then click on Blank, then the area under "Issue #119" doesn't get updated.

If you click any yellow box except SQL, then click SQL, then any other yellow box, then the area under "Issue Other" doesn't get updated ever again (except perhaps for SQL).

The "Issue #119" area is about just highlighting code, and handling a blank value.
The "Issue Other" area is about whether or not to highlight the code (and if not, then showing the unhighlighted code).

This affects 4.1.3 with Angular 11, too.

**Comment by cbejensen** [comment]
I am experiencing this issue, but it's easily solvable:

```html
<code [highlight]="value || ' '"></code>
```

That being said, I think the content should be updated even if it's `null` or `undefined`.

**Comment by estanglerbm** [comment]
I know (my stackblitz illustrated the space workaround, too), but there may be cases where a space character is not desired.

But the real issue here, as I previously mentioned, is that there are two use cases:  one where you want nothing to happen, and one where you want to clear the area.  There should be a mechanism to do both things (and a highlight="" would probably do one of them, and another mechanism does the other, or maybe solve the situation differently).

Otherwise, you're just going to continue to have future commits conflicting on this, because some people expect highlight="" to do nothing, and others expect highlight="" to clear the area.

**Comment by MurhafSousli** [comment]
@estanglerbm I would recommend listing all the use cases and their expected behaviour as bullet points to easily follow on them. I find it confusing to read the use cases and how it should behave from the comment or from the stackblitz.

**Comment by FrancescoBorzi** [comment]
@MurhafSousli I'd recommend a test case which will prevent this issue from being introduced again by mistake

**Comment by MurhafSousli** [comment]
Ok, it is fixed in 6.1.1 and I added test cases for that.

Now the expected behavior is the following:

- If you pass a null value, nothing is gonna change, it will not highlight, and it will not change a previously highlighted code.
- If you pass an empty string, it will clear the highlighted content.

**Comment by estanglerbm** [comment]
@MurhafSousli Fantastic.  (I can't test it for a while, but the behavior sounds good.)  Thanks.
